### PR TITLE
Added missing correlation id

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
Fixes # 
Missing correlationId

**Changes proposed in this request**
Added the missing Id

**Testing**
NA. Minor

**Performance impact**
None

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs [hunk 1]
```diff
@@ -120,6 +120,7 @@ internal static MsalTokenResponse CreateFromiOSBrokerResponse(Dictionary<string,
                     AccountUserId = homeAcctId != null ? AccountId.ParseFromString(homeAcctId).ObjectId : null,
                     TenantId = homeAcctId != null ?  AccountId.ParseFromString(homeAcctId).TenantId : null,
                     Upn = (metadataDictionary?.ContainsKey(TokenResponseClaim.Upn) ?? false) ? metadataDictionary[TokenResponseClaim.Upn] : null,
+                    CorrelationId = responseDictionary[BrokerResponseConst.CorrelationId],
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:123]: Are you sure it's always there? Maybe it's worth doing a `ContainsKey` check?

File: src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs [hunk 2]
```diff
@@ -25,8 +25,10 @@ internal static string GetEnrollmentId(ICoreLogger logger)
                 {
                     var enrollmentIDs = JsonConvert.DeserializeObject<EnrollmentIDs>(keychainData);
 
-                    return enrollmentIDs.EnrollmentIds[0].EnrollmentId;
-
+                    if (enrollmentIDs.EnrollmentIds.Count > 0)
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs:28]: possible null refs?

File: src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs (hunk 1 above)
Comment (SameerK-MSFT) [reply src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:123]: Fixed with other PR. https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3399/

**Comment by SameerK-MSFT** [comment]
Superset of this branch is now with another branch. Ref PR: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3399/

**Comment by SameerK-MSFT** [comment]
Updated in other branch as mentioned above. Closing.

**Pull request closed without merging**